### PR TITLE
feat(spanner): add MR CMEK samples

### PR DESCRIPTION
**Pull request by panerorenn9541**
Add samples and tests for create database, create backup, copy backup, and restore backup.

### CODE REVIEW

File: spanner/src/copy_backup_with_MR_CMEK.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ * @param string $sourceInstanceId The Spanner instance ID of the source backup.
+ * @param string $sourceBackupId The Spanner backup ID of the source.
+* @param string[] $kmsKeyNames The KMS keys used for encryption.
+ */
+function copy_backup_with_MR_CMEK(
+    string $projectId,
+    string $destInstanceId,
+    string $destBackupId,
+    string $sourceInstanceId,
+    string $sourceBackupId
```
Comment (Hectorhammett) [inline spanner/src/copy_backup_with_MR_CMEK.php:63]: Missing `,`

File: spanner/src/copy_backup_with_MR_CMEK.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        printf(
+            'Backup %s of size %d bytes was copied at %d from the source backup %s using encryption keys %s' . PHP_EOL,
+            basename($destBackupInfo->getName()),
+            $destBackupInfo->getSizeBytes(),
+            $destBackupInfo->getCreateTime()->getSeconds(),
+            $sourceBackupId,
+            print_r($info->getEncryptionInfo()->getKmsKeyVersions(), true)
+        );
+        printf('Version time of the copied backup: %d' . PHP_EOL, $destBackupInfo->getVersionTime()->getSeconds());
+    } else {
```
Comment (Hectorhammett) [inline spanner/src/copy_backup_with_MR_CMEK.php:97]: This is just a nit, but I always believe that is best to handle errors before the "main logic" or "happy path" if you will:
```php
if (!operationResponse->operationSucceeded()) {
     $error = $operationResponse->getError();
     printf('Backup not created due to error: %s.' . PHP_EOL, $error->getMessage());
     return;
}
...
```

That way the happy path and what the code is aiming to perform is on the main indentation and easier to glance at.

**Review by Hectorhammett — CHANGES_REQUESTED** [review]

**Comment by Hectorhammett** [comment]
Looking good but there are some errors on the syntax and style on the tests :).

**Review by Hectorhammett — APPROVED** [review]

**Review by Hectorhammett — APPROVED** [review]

File: spanner/src/copy_backup_with_MR_CMEK.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ * ```
+ *
+ * @param string $projectId The Google Cloud project ID.
+ * @param string $destInstanceId The Spanner instance ID where the copy backup will reside.
+ * @param string $destBackupId The Spanner backup ID of the new backup to be created.
+ * @param string $sourceInstanceId The Spanner instance ID of the source backup.
+ * @param string $sourceBackupId The Spanner backup ID of the source.
+* @param string[] $kmsKeyNames The KMS keys used for encryption.
+ */
+function copy_backup_with_MR_CMEK(
```
Comment (bshaffer) [inline spanner/src/copy_backup_with_MR_CMEK.php:58]: nit: all sample names (and filenames) are lowercase in this repo. Let's be consistent!

File: spanner/src/create_backup_with_MR_CMEK.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/tree/main/spanner/README.md
+ */
+
+namespace Google\Cloud\Samples\Spanner;
+
+// [START spanner_create_backup_with_MR_CMEK]
```
Comment (bshaffer) [inline spanner/src/create_backup_with_MR_CMEK.php:26]: Same with sample region tags

Comment (panerorenn9541) [reply spanner/src/create_backup_with_MR_CMEK.php:26]: The tags I got from Tina and we are using the same tags in all languages

**Review by bshaffer — APPROVED** [review]

**Pull request merged**